src/MatrixClientPeg.ts

```diff
@@ -212,12 +213,22 @@ class MatrixClientPegClass implements IMatrixClientPeg {
             // If the user is not a guest then prompt them to reload rather than doing it for them
             // For guests this is likely to happen during e-mail verification as part of registration
 
-            const { finished } = Modal.createDialog(ErrorDialog, {
-                title: _t("error_database_closed_title"),
-                description: _t("error_database_closed_description"),
-                button: _t("action|reload"),
-            });
-            const [reload] = await finished;
+            const ua = navigator.userAgent;
+            const deviceType = parseUserAgent(ua).deviceType;
```

Comment:
We shouldn't be relying on UA parsing, you can use the Platforms for this. WebPlatform vs ElectronPlatform.

Reply:
> We shouldn't be relying on UA parsing, you can use the Platforms for this. WebPlatform vs ElectronPlatform.

I have made changes, PTAL 👍